src/lib.rs

```diff
@@ -213,108 +185,234 @@ impl Build {
+                "kqueue",
+                "lb",
+                "mailbox_safe",
+                "mailbox",
+                "mechanism_base",
+                "mechanism",
+                "metadata",
+                "msg",
+                "mtrie",
+                "norm_engine",
+                "null_mechanism",
+                "object",
+                "options",
+                "own",
+                "pair",
+                "peer",
+                "pgm_receiver",
+                "pgm_sender",
+                "pgm_socket",
+                "pipe",
+                "plain_client",
+                "plain_server",
+                "poll",
+                "poller_base",
+                "polling_util",
+                "pollset",
+                "precompiled",
+                "proxy",
+                "pub",
+                "pull",
+                "push",
+                "radio",
+                "radix_tree",
+                "random",
+                "raw_decoder",
+                "raw_encoder",
+                "raw_engine",
+                "reaper",
+                "rep",
+                "req",
+                "router",
+                "scatter",
+                "select",
+                "server",
+                "session_base",
+                "signaler",
+                "socket_base",
+                "socket_poller",
+                "socks_connecter",
+                "socks",
+                "stream_connecter_base",
+                "stream_engine_base",
+                "stream_listener_base",
+                "stream",
+                "sub",
+                "tcp_address",
+                "tcp_connecter",
+                "tcp_listener",
+                "tcp",
+                "thread",
+                "timers",
+                "tipc_address",
+                "tipc_connecter",
+                "tipc_listener",
+                "trie",
+                "udp_address",
+                "udp_engine",
+                "v1_decoder",
+                "v1_encoder",
+                "v2_decoder",
+                "v2_encoder",
+                "v3_1_encoder",
+                "vmci_address",
+                "vmci_connecter",
+                "vmci_listener",
+                "vmci",
+                "ws_address",
+                "ws_connecter",
+                "ws_decoder",
+                "ws_encoder",
+                "ws_engine",
+                "ws_listener",
+                // "wss_address", // requires gnutls
+                // "wss_engine", // requires gnutls
+                "xpub",
+                "xsub",
+                "zap_client",
+                "zmq_utils",
+                "zmq",
+                "zmtp_engine",
+            ],
+        );
+
+        add_c_sources(&mut build, path.join("external/sha1"), &["sha1.c"]);
+
+        build.define("ZMQ_USE_CV_IMPL_STL11", "1");
+        build.define("ZMQ_BUILD_DRAFT_API", "1");
+        build.define("ZMQ_STATIC", "1");
```

Comment:
I did this mostly because the upstream `zmq-sys` crate just hardcodes it to `true` regardless. Figured that path wasn't taken anymore and could be removed.

Reply:
I think it makes sense to always hardcode to static and remove the static feature because we're vendoring the dependency anyway.